src/tests/test_magic.py

```diff
@@ -1305,3 +1306,82 @@ def test_interact_and_missing_ipywidgets_installed(ip):
             "%sql --interact my_variable SELECT * FROM author LIMIT {{my_variable}}"
         )
         assert isinstance(out.error_in_exec, ModuleNotFoundError)
+
+
+def test_save_snippet_as_sql(ip):
```

Comment:
I think we should pass `tmp_empty` fixture to prevent files from getting created in the original directory.

Reply:
I have created a fixture `ip_snip` for the tests related to saving snippets as .sql files, incorporated tmp_empty here